starlingbank/__init__.py

```diff
@@ -212,6 +248,51 @@ def _set_basic_account_data(self):
         self.currency = account["currency"]
         self.created_at = account["createdAt"]
 
+    def update_spending_categories_data(self, month: str = None, year: str = None) -> None:
+        """Get the spending categories information for the account, default for the latest month."""
+        if not month or not year:
+            year = str(datetime.now().year)
+            month = datetime.now().strftime("%B").upper()  # Converts month to uppercase
+
+        response = get(
+            _url(
+                "/accounts/{0}/spending-insights/spending-category?year={1}&month={2}".format(self._account_uid,
+                                                                                              year, month),
+                self._sandbox,
+            ),
+            headers=self._auth_headers,
+        )
+        response.raise_for_status()
+
+        response = response.json()
+        response_spending_categories = response.get("breakdown", {})
+
```

Comment:
I think the logic below can be simplified to:

1. Clear the existing `self.spending_categories` dict.
2. Create a `SpendingCategory` instance for each item in `response_spending_categories`.
3. Add each instance of `SpendingCategory` to `self.spending_categories`.

I don't think there's a need to loop through and dynamically create/update/clear. This also negates the need for the `update` method above.

Reply:
I modelled this on the `update_savings_goal_data` method. 
My interpretation was that you allow a user to pass in a list of savings goals and the method can be used to loop through the user provided list. I want to allow a user to pass in a list of spending categories so I think I need a method to loop through this user provider list and handle any mismatches between the user list and the API response.
Hence this code:
```
        # Set values to zero if category not returned
        for category_name in self.spending_categories:
            if category_name not in returned_categories:
                self.spending_categories[category_name].update({
                    "totalSpent": 0.0,
                    "totalReceived": 0.0,
                    "netSpend": 0.0,
                    "percentage": 0.0,
                    "transactionCount": 0,
                })
```

Reply:
This is old code, and I don't think I'd repeat the pattern I implemented today. But I don't have the appetite to rewrite it, so best to stick with the existing patten.

Essentially, `update_savings_goal_data()` will sync known savings goals (`self.savings_goals`). Then it calls the `update` method on each of these savings goals and, as it already has the data, it passes it in the update call. Crucially, if no `goal` data is supplied to the `update` method, the savings goal will update itself individually.

The difference with spending categories is that they can't update themselves individually. You can only sync all. So to my mind it makes sense to keep things simple and just replace the entire `self.spending_categories` array with the new data. No need for the individual update methods.

So my proposal is for `SpendingCategory` to be a simple data class with just an `__init__` method.

I hope this all makes sense.